### PR TITLE
Create our own Enumerable

### PR DESCRIPTION
**Pull request by anita00001**
# In this PR, @anita00001 and @Othniel2471 have collaborated to:

### Create module
- Created module `MyEnumerable` in file `my_enumerable.rb`
    - `MyEnumerable` will implement a subset of the functionality of [`Enumerable`](https://ruby-doc.org/core-3.0.0/Enumerable.html)

#### Create methods in the module `MyEnumerable`
- Created the `my_all?` method that resembles the functionality of the `all?` method in Ruby
- Created the `my_any?` method that resembles the functionality of the `any?` method in Ruby
- Created the `my_filter` method that resembles the functionality of the `filter` method in Ruby

### Create class
- Created a class `MyList` in the `file my_list.rb` that has an instance variable `@list`
- Used the `MyEnumerable` module in class `MyList`
- Implemented a method `#each` in class `MyList` that yields successive members of `@list`

### Verify solution
- Created an instance of a class `MyList` with some initial values in file list_test_case.rb
- Tested `my_all?` method
- Tested `my_any?` method
- Tested `my_filter` method

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Team

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have a descriptive and professional readme 👍🏿 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -58,234 +17,143 @@ For more information on the importance of a professional README for your reposit
...
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-<!-- LICENSE -->
-
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
+> This project is [MIT](./MIT) licensed.
```
Comment (V-Blaze) [inline README.md]: - You have done great so far :+1: Your project works as expected. But there are a few requirements you need to meet before moving to the next milestone. :smile: 
- [ ] Currently, the MIT link in your readme file leads to a 404 error page. I'll suggest you fix this as broken links on readme makes the repository look a bit unprofessional. :+1: :small_airplane: :wave: 

- - Screenshot:
Screenshot from 2023-08-24 15-58-52

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def each(&block)
+    @list.each(&block)
+  end
+end
+
+# list.my_filter {|e| e.even?}
+
+# list.my_any? {|e| e == 2}
+
+# list.my_any? {|e| e == 5}
```
Comment (V-Blaze) [inline my_list.rb]: - [ ] It's against the best practices to leave un-used code as a comment on your code base, I'll suggest you remove this. :smile:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @anita00001, @Othniel2471  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


## Highlights
- used `require_relative` to import `Enumerable`✔✔
- Implemented all Enumerable methods `#all?` `#any?` and `#filter` correctly :heavy_check_mark: 
- All test cases passed :heavy_check_mark: 

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — APPROVED** [review]
Hi @anita00001 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**